packages/core/src/render3/instructions.ts

```diff
@@ -1557,6 +1552,7 @@ function valueInData<T>(data: any[], index: number, value?: T): T {
   if (value === undefined) {
     value = data[index];
   } else {
+    ngStaticData[index] = null;
```

Comment:
we should not write unless we are sure that the previous value is not the correct one.

Reply:
Right, we should check if the index is >= the length of static data to see if it's been created yet.  If it hasn't, we should set to null because locals don't store static data. Fixed.